### PR TITLE
upload job related fixes (WP-770)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Submissions/SubmissionManager.php [hunk 1]
```diff
@@ -722,7 +722,10 @@ private function getConditionBlockForUploadJob(): ConditionBlock
         $block = new ConditionBlock(ConditionBuilder::CONDITION_BLOCK_LEVEL_OPERATOR_AND);
         $block->addCondition(Condition::getCondition(ConditionBuilder::CONDITION_SIGN_EQ, SubmissionEntity::FIELD_STATUS, [SubmissionEntity::SUBMISSION_STATUS_NEW]));
         $block->addCondition(Condition::getCondition(ConditionBuilder::CONDITION_SIGN_EQ, SubmissionEntity::FIELD_IS_LOCKED, [0]));
-        $block->addCondition(Condition::getCondition(ConditionBuilder::CONDITION_SIGN_NOT_EQ, SubmissionEntity::FIELD_BATCH_UID, ['']));
+        $innerBlock = new ConditionBlock(ConditionBuilder::CONDITION_BLOCK_LEVEL_OPERATOR_OR);
+        $innerBlock->addCondition(Condition::getCondition(ConditionBuilder::CONDITION_SIGN_NOT_EQ, SubmissionEntity::FIELD_BATCH_UID, ['']));
+        $innerBlock->addCondition(Condition::getCondition(ConditionBuilder::CONDITION_SIGN_EQ, SubmissionEntity::FIELD_IS_CLONED, [1]));
+        $block->addConditionBlock($innerBlock);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Submissions/SubmissionManager.php:728]: Fix visibility of cloned items in upload job settings table

File: inc/Smartling/Helpers/EntityHelper.php [hunk 2]
```diff
@@ -86,10 +86,7 @@ public function getLogger(): LoggerInterface
         return $this->logger;
     }
 
-    /**
-     * @return SettingsManager
-     */
-    public function getSettingsManager()
+    public function getSettingsManager(): SettingsManager
```
Comment (vsolovei-smartling) [inline inc/Smartling/Helpers/EntityHelper.php:89]: Simplify testing. Will refactor code to delete EntityHelper entirely

File: inc/Smartling/WP/Table/SubmissionTableWidget.php [hunk 3]
```diff
@@ -178,10 +184,10 @@ private function processBulkAction(): void
             return (int)$value;
         }, $this->getFormElementValue('submission', []));
 
-        if (is_array($submissionsIds) && 0 < count($submissionsIds)) {
```
Comment (vsolovei-smartling) [inline inc/Smartling/WP/Table/SubmissionTableWidget.php:181]: $submissionsIds = array_map, so always an array

File: inc/Smartling/WP/Table/SubmissionTableWidget.php [hunk 4]
```diff
@@ -205,11 +211,12 @@ private function processBulkAction(): void
                             ];
                             $this->queue->enqueue([$submission->getId()], QueueInterface::QUEUE_NAME_DOWNLOAD_QUEUE);
                         }
-                        if ($profile === null) {
+                        if ($profile !== null) {
```
Comment (vsolovei-smartling) [inline inc/Smartling/WP/Table/SubmissionTableWidget.php:214]: Not task related, just something I found out while working on it, audit log records not created when downloading from bulk screen

File: inc/Smartling/WP/Table/SubmissionTableWidget.php [hunk 5]
```diff
@@ -386,9 +393,8 @@ public function prepare_items(): void
             $row[SubmissionEntity::FIELD_FILE_URI] = $fileName;
             $row[SubmissionEntity::FIELD_SOURCE_TITLE] = htmlentities($row[SubmissionEntity::FIELD_SOURCE_TITLE]);
             $row[SubmissionEntity::FIELD_CONTENT_TYPE] = WordpressContentTypeHelper::getLocalizedContentType($row[SubmissionEntity::FIELD_CONTENT_TYPE]);
-            $row[SubmissionEntity::FIELD_SUBMISSION_DATE] = DateTimeHelper::toWordpressLocalDateTime(DateTimeHelper::stringToDateTime($row[SubmissionEntity::FIELD_SUBMISSION_DATE]));
-            $row[SubmissionEntity::FIELD_APPLIED_DATE] = '0000-00-00 00:00:00' === $row[SubmissionEntity::FIELD_APPLIED_DATE] ? __('Never')
-                : DateTimeHelper::toWordpressLocalDateTime(DateTimeHelper::stringToDateTime($row[SubmissionEntity::FIELD_APPLIED_DATE]));
+            $row[SubmissionEntity::FIELD_SUBMISSION_DATE] = $this->sqlToReadableDate($row[SubmissionEntity::FIELD_SUBMISSION_DATE]);
+            $row[SubmissionEntity::FIELD_APPLIED_DATE] = $this->sqlToReadableDate($row[SubmissionEntity::FIELD_APPLIED_DATE]);
```
Comment (vsolovei-smartling) [inline inc/Smartling/WP/Table/SubmissionTableWidget.php:397]: Fix invalid date display in submissions table

File: inc/Smartling/Submissions/SubmissionManager.php [hunk 6]
```diff
@@ -137,7 +137,11 @@ private function getTotalByFieldInValues(ConditionBlock $conditionBlock): int
 
     public function getTotalInUploadQueue(): int
     {
-        return $this->getTotalByFieldInValues($this->getConditionBlockForUploadJob());
+        $block = new ConditionBlock(ConditionBuilder::CONDITION_BLOCK_LEVEL_OPERATOR_OR);
+        $block->addConditionBlock($this->getConditionBlockForCloning());
+        $block->addConditionBlock($this->getConditionBlockForUploadJob());
+
+        return $this->getTotalByFieldInValues($block);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Submissions/SubmissionManager.php:144]: Add visibility of items that need to be cloned in the list table

File: inc/Smartling/WP/Table/SubmissionTableWidget.php [hunk 7]
```diff
@@ -169,6 +167,14 @@ public function get_bulk_actions(): array
         ];
     }
 
+    /**
+     * Testing wrapper
+     * @return false|string
+     */
+    public function getCurrentAction() {
+        return $this->current_action();
+    }
```
Comment (PavelLoparev) [inline inc/Smartling/WP/Table/SubmissionTableWidget.php:176]: Why can't you mock `current_action` directly?

File: inc/Smartling/WP/Table/SubmissionTableWidget.php (hunk 4 above)
Comment (PavelLoparev) [inline inc/Smartling/WP/Table/SubmissionTableWidget.php:214]: :+1:

File: inc/Smartling/WP/Table/SubmissionTableWidget.php (hunk 7 above)
Comment (vsolovei-smartling) [reply inc/Smartling/WP/Table/SubmissionTableWidget.php:176]: It's a wordpress function, if we will improve testing, a wrapper is preferential

Comment (vsolovei-smartling) [reply inc/Smartling/WP/Table/SubmissionTableWidget.php]: Removed redundant code, thanks!

**Review by PavelLoparev — APPROVED** [review]

**Review by dimitrystd — APPROVED** [review]

**Pull request merged**